src/video/SDL_video.c

```diff
@@ -1435,6 +1435,11 @@ void SDL_SetDesktopDisplayMode(SDL_VideoDisplay *display, const SDL_DisplayMode
 {
     SDL_DisplayMode last_mode;
 
+    if (display->fullscreen_window) {
```

Comment:
```suggestion
    if (display->fullscreen_window || display->device->setting_display_mode)
```

When entering fullscreen, the fullscreen window pointer isn't set until after the mode switch, so the display mode setting flag should be checked too. This allows for reverting b69201daebfe0e7e0e59cd5fe0f4a24a07d5ae60 since the X and Mac cases can both handled here.